gc/stats/ScavengerHotFieldStats.cpp

```diff
@@ -27,12 +27,34 @@
 
 #if defined(OMR_GC_MODRON_SCAVENGER) || defined(OMR_GC_VLHGC)
 
+MM_ScavengerHotFieldStats*
+MM_ScavengerHotFieldStats::newInstance(MM_GCExtensionsBase *extensions)
+{
+	MM_ScavengerHotFieldStats *scavengerHotFieldStats;
+
+	scavengerHotFieldStats = (MM_ScavengerHotFieldStats *)extensions->getForge()->allocate(sizeof(MM_ScavengerHotFieldStats), OMR::GC::AllocationCategory::FIXED, OMR_GET_CALLSITE());
+	if (NULL != scavengerHotFieldStats) {
+		new(scavengerHotFieldStats) MM_ScavengerHotFieldStats();
+		if (!scavengerHotFieldStats->initialize(extensions)) {
+			scavengerHotFieldStats->kill(extensions);
+			scavengerHotFieldStats = NULL;
+		}
+	}
+	return scavengerHotFieldStats;
+}
+
 bool
-MM_ScavengerHotFieldStats::initialize(MM_EnvironmentBase *env)
+MM_ScavengerHotFieldStats::initialize(MM_GCExtensionsBase *extensions)
 {
-	_objectModel = &(env->getExtensions()->objectModel);
+	_objectModel = &(extensions->objectModel);
 	return true;
 }
 
+void
+MM_ScavengerHotFieldStats::kill(MM_GCExtensionsBase *extensions)
+{
+	extensions->getForge()->free(this);

```

Comment:
here we should call tearDown() even it does nothing

Reply:
👍